components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -24,12 +25,35 @@ object TypeConversionHandler {
   // TODO: Add feature flag: strictBigDecimalChecking (default false?) and rename strictTypeChecking to strictClassesTypeChecking
   private val ConversionFromClassesForDecimals = NumberTypesPromotionStrategy.DecimalNumbers.toSet + classOf[java.math.BigDecimal]
 
-  private val ValueClassesThatBeConvertedFromString = List[Class[_]](classOf[ZoneId], classOf[ZoneOffset], classOf[Locale], classOf[Charset], classOf[Currency], classOf[UUID],
-    classOf[LocalTime], classOf[LocalDate], classOf[LocalDateTime], classOf[ChronoLocalDate], classOf[ChronoLocalDateTime[_]])
+  val stringConversionsMap: Map[Class[_], String => Object] = Map(
```

Comment:
Can you do this type-safety so it won't be possible to write conversion logic returning another type than in key?

Reply:
I don't really have an idea how to do this to be honest